src/grisp_i2c.erl

```diff
@@ -26,9 +26,8 @@
 start_link(DriverMod) ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, DriverMod, []).
 
-msgs(Msgs) ->
-    Enc_msgs = encode_msgs(Msgs),
-    gen_server:call(?MODULE, {msgs, Enc_msgs}).
+msgs([Adr | Msgs]) ->
+    do_msgs(Adr, Msgs).
```

Comment:
The message handling must be done inside the `gen_server` process of the I₂C driver. Otherwise the message sequence would be interleaved when multiple callers are sending messages. The encode should still be done in the caller process since it is a static operation that doesn't have to be run in the `gen_server`.